driver-core/src/main/com/mongodb/internal/capi/MongoCryptHelper.java

```diff
@@ -111,6 +112,9 @@ public static BsonDocument fetchCredentials(final Map<String, Map<String, Object
                 kmsProvidersDocument.put("aws", awsCredentialDocument);
             }
         }
+        if (kmsProvidersDocument.containsKey("gcp") && kmsProvidersDocument.get("gcp").asDocument().isEmpty()) {
```

Comment:
This is the hook to get the GCP creds from the environment.  Similar patter to AWS above.